src/query/storages/system/src/tables_table.rs

```diff
@@ -245,7 +250,28 @@ where TablesTable<T, U>: HistoryAware
                                     db_name.push(database.clone());
                                 }
                             }
+                        } else if col_name == "table_id" {
+                            match check_number::<_, u64>(
+                                None,
+                                &FunctionContext::default(),
+                                &Expr::<usize>::Constant {
+                                    span: None,
+                                    scalar: scalar.clone(),
+                                    data_type: scalar.as_ref().infer_data_type(),
+                                },
+                                &BUILTIN_FUNCTIONS,
+                            ) {
+                                Ok(id) => tables_ids.push(id),
+                                Err(_) => invalid_tables_ids = true,
```

Comment:
If the table id is not u64, the query conditions are not met. Can we return an empty Block?

Reply:
No. maybe this : name = 'xx' or table_id = 123.1.

Reply:
ok